app/src/main/kotlin/com/saltedge/authenticator/widget/security/UnlockAppInputView.kt

```diff
@@ -55,22 +67,31 @@ class UnlockAppInputView(context: Context, attrs: AttributeSet) : LinearLayout(c
         )
     }
 
-    fun setDescriptionText(textResId: Int) {
-        descriptionView?.setText(textResId)
-    }
-
     fun setDescriptionText(text: String) {
         descriptionView?.text = text
     }
 
+    fun setDescriptionText(textResId: Int) {
+        descriptionView?.setText(textResId)
+    }
+
     fun setInputViewVisibility(show: Boolean) {
         passcodeInputView?.setVisible(show)
+        appLogoView?.setVisible(show)
     }
 
-    private fun updateDescription(biometricsActionIsAvailable: Boolean) {
-        descriptionView?.setText(
-            if (biometricsActionIsAvailable) R.string.actions_enter_passcode_or_fingerprint
-            else R.string.actions_enter_passcode
-        )
+    fun setResetPasscodeView() {
```

Comment:
why you setup views in arbitrary time in public function, but not on view init as usual?